orderer/consensus/etcdraft/membership.go

```diff
@@ -105,6 +114,102 @@ func ComputeMembershipChanges(oldMetadata *etcdraft.BlockMetadata, oldConsenters
 	return result, nil
 }
 
+func validateConsenterTLSCerts(c *etcdraft.Consenter, ordererConfig channelconfig.Orderer) error {
+	clientCert, err := parseCertificateFromBytes(c.ClientTlsCert)
+	if err != nil {
+		return errors.Wrap(err, "parsing tls client cert")
+	}
+	serverCert, err := parseCertificateFromBytes(c.ServerTlsCert)
+	if err != nil {
+		return errors.Wrap(err, "parsing tls server cert")
+	}
+
+	for _, org := range ordererConfig.Organizations() {
+		opts, err := createX509VerifyOptions(org.MSP())
+		if err != nil {
+			return errors.WithMessage(err, "creating x509 verify options")
+		}
+
+		// following the lead of msp/mspimplvalidate.go:
+		// This is a "temporary" workaround for FAB-3678
+		opts.CurrentTime = clientCert.NotBefore.Add(time.Second)
+		_, err = clientCert.Verify(opts)
+		if err != nil {
+			// if client cert does not have a validation chain for this org,
+			// continue to next organization
+			continue
+		}
+
+		// again following the lead of msp/mspimplvalidate.go:
+		// This is a "temporary" workaround for FAB-3678
+		opts.CurrentTime = serverCert.NotBefore.Add(time.Second)
+		_, err = serverCert.Verify(opts)
+		if err == nil {
+			return nil
+		}
+	}
+
+	return errors.New("consenter client/server tls certs are not valid for any existing orderer organizations")
+}
+
+func createX509VerifyOptions(m msp.MSP) (x509.VerifyOptions, error) {
+	rootCerts, err := parseCertificateListFromBytes(m.GetTLSRootCerts())
+	if err != nil {
+		return x509.VerifyOptions{}, errors.Wrap(err, "parsing tls root certs")
+	}
+	intermediateCerts, err := parseCertificateListFromBytes(m.GetTLSIntermediateCerts())
+	if err != nil {
+		return x509.VerifyOptions{}, errors.Wrap(err, "parsing tls intermediate certs")
+	}
+
+	tlsRoots := x509.NewCertPool()
+	for _, cert := range rootCerts {
+		tlsRoots.AddCert(cert)
+	}
+	tlsIntermediates := x509.NewCertPool()
+	for _, cert := range intermediateCerts {
+		tlsIntermediates.AddCert(cert)
+	}
+
+	return x509.VerifyOptions{
+		Roots:         tlsRoots,
+		Intermediates: tlsIntermediates,
+		KeyUsages: []x509.ExtKeyUsage{
```

Comment:
what if we only have one of the two key usages? won't this fail?

Reply:
No, this accepts certificates as long as they have any of the key usages specified.

Reply:
Done.

Reply:
Thinking a little more while I'm working on the negative integration test: do/should we verify the key usages separately, i.e. check that the ServerTlsCert has the server key usage and the ClientTlsCert has the client key usage?

Reply:
I don't think so, there is a limit to which we need to coddle the users.